From ba9c476c558647b666ea93dda72ff40d6a6e9505 Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Tue, 12 Mar 2024 18:40:44 -0500
Subject: [PATCH] LFU-641: capsule: Fix build break due to bashism

Fixes:
```
| NOTE: building iMX8MP - TEE=tee.bin-stmm flash_evk_stmm_capsule
| ls: cannot access 'CRT.*': No such file or directory
| ls: cannot access 'CRT.*': No such file or directory
| ls: cannot access 'CRT.*': No such file or directory
| ****************************************************************
| Key  already existed
| If you not wanna use new Key, please not run target: capsule_key
| Otherwise, please delete CRT.* and re-run capsule_key
| ****************************************************************
| make[1]: *** [soc.mak:259: capsule_key] Error 1
```

Signed-off-by: Tom Hochstein <tom.hochstein@nxp.com>
---
 iMX8M/soc.mak | 6 +++---
 iMX9/soc.mak  | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index 0ff1dff..6751759 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -247,11 +247,11 @@ flash_ddr4_val: flash_ddr4_val_no_hdmi
 endif
 
 ifeq ($(TEE),tee.bin-stmm)
-KEY_EXISTS = $(shell if ls CRT.* &> /dev/null 2>&1; then echo "exist"; else echo "noexist"; fi)
+KEY_EXISTS = $(shell find -maxdepth 1 -name "CRT.*")
 capsule_key:
-ifeq ($(KEY_EXISTS),exist)
+ifneq ($(KEY_EXISTS),)
 	@echo "****************************************************************"
-	@echo "Key $(shell ls CRT.*) already existed"
+	@echo "Key $(KEY_EXISTS) already existed"
 	@echo "If you not wanna use new Key, please not run target: capsule_key"
 	@echo "Otherwise, please delete CRT.* and re-run capsule_key"
 	@echo "****************************************************************"
diff --git a/iMX9/soc.mak b/iMX9/soc.mak
index b17db6b..13a7891 100644
--- a/iMX9/soc.mak
+++ b/iMX9/soc.mak
@@ -249,11 +249,11 @@ clean:
 
 # Add for System ready
 ifeq ($(TEE),tee.bin-stmm)
-KEY_EXISTS = $(shell if ls *CRT.* &> /dev/null 2>&1; then echo "exist"; else echo "noexist"; fi)
+KEY_EXISTS = $(shell find -maxdepth 1 -name "CRT.*")
 capsule_key:
-ifeq ($(KEY_EXISTS),exist)
+ifneq ($(KEY_EXISTS),)
 	@echo "****************************************************************"
-	@echo "Key $(shell ls CRT.*) already existed"
+	@echo "Key $(KEY_EXISTS) already existed"
 	@echo "If you not wanna use new Key, please not run target: capsule_key"
 	@echo "Otherwise, please delete CRT.* and re-run capsule_key"
 	@echo "****************************************************************"
-- 
2.25.1

